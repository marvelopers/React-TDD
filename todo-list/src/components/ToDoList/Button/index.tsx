import React from 'react';
import styled, { css } from 'styled-components';

interface ButtonProps {
  text: string;
  backgroundColor?: string;
  hoverColor?: string;
  onClick: () => void;
}

const Button = ({ text, backgroundColor = '#304FFE', hoverColor = '#1E40FF', onClick }: ButtonProps) => {
  return (
    <SingleButton backgroundColor={backgroundColor} hoverColor={hoverColor} onClick={onClick}>
      {text}
    </SingleButton>
  );
};

export default Button;

const SingleButton = styled.button(
  ({ backgroundColor, hoverColor }: { backgroundColor: string; hoverColor: string }) => css`
    padding: 10px 20px;
    background-color: ${backgroundColor};
    border-radius: 8px;
    border: none;
    outline: none;
    text-align: center;

    color: #dbf6fd;
    cursor: pointer;

    :hover {
      background-color: ${hoverColor};
    }
  `,
);
