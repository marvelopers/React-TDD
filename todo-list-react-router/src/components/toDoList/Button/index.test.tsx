import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import 'jest-styled-components';

import Button from './index';

describe('<Button/>', () => {
  it('renders components correctly', () => {
    const { container } = render(<Button text="TEST" onClick={() => jest.fn()} />);

    const text = screen.getByText('TEST');
    expect(text).toBeInTheDocument();

    expect(text).toHaveStyleRule('background-color', '#304FFE');
    // expect(text).toHaveStyleRule('background-color:', '#1E40FF', { modifier: ':hover' });

    expect(container).toMatchSnapshot();
  });

  it('Clicks the button', () => {
    const handleClick = jest.fn();
    render(<Button text="TEST" onClick={handleClick} />);

    const text = screen.getByText('TEST');
    expect(handleClick).toHaveBeenCalledTimes(0);
    fireEvent.click(text);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
