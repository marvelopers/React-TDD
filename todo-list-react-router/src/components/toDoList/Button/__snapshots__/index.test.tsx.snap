// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> renders components correctly 1`] = `
.c0 {
  padding: 10px 20px;
  background-color: #304FFE;
  border-radius: 8px;
  border: none;
  outline: none;
  text-align: center;
  color: #dbf6fd;
  cursor: pointer;
}

.c0:hover {
  background-color: #1E40FF;
}

<div>
  <button
    class="c0"
  >
    TEST
  </button>
</div>
`;
